spring-integration-mail/src/test/java/org/springframework/integration/mail/TestMailServer.java

```diff
@@ -282,10 +282,10 @@ else if (line.contains("FETCH 1 (ENVELOPE")) {
 							write("* 1 FETCH (RFC822.SIZE 6909 INTERNALDATE \"27-May-2013 09:45:41 +0000\" "
 									+ "FLAGS (\\Seen) "
 									+ "ENVELOPE (\"Mon, 27 May 2013 15:14:49 +0530\" "
-									+ "\"Test Email\" ((\"Foo\" NIL \"foo\" \"bar.tv\")) "
+									+ "\"Test Email\" ((\"Bar\" NIL \"bar\" \"baz\")) "
 									+ "((\"Foo\" NIL \"foo\" \"bar.tv\")) "
 									+ "((\"Foo\" NIL \"foo\" \"bar.tv\")) "
-									+ "((\"Bar\" NIL \"bar\" \"baz.net\")) NIL NIL "
+									+ "((\"Foo\" NIL \"foo\" \"bar\")) NIL NIL "
```

Comment:
Yep! Makes sense.
The problem that my tests in Java DSL had already a code to test content. So, my first impression was that we have something wrong in the `HeaderMapper` process when my tests have started to fail.